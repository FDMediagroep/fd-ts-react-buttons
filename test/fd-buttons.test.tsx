import TestRenderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils';
import Button from '../src/fd-buttons';
import React from 'react';

describe('Button', () => {
    test('renders correctly', () => {
        let button = TestRenderer.create(<Button/>);
        expect(button.toJSON()).toMatchSnapshot();
        button = TestRenderer.create(<Button type="submit"/>);
        expect(button.toJSON()).toMatchSnapshot();
        button = TestRenderer.create(<Button type="button" className="test1"/>);
        expect(button.toJSON()).toMatchSnapshot();
        button = TestRenderer.create(<Button type="submit" className="test2"/>);
        expect(button.toJSON()).toMatchSnapshot();
    });

    test('renders primary correctly', () => {
        let button = TestRenderer.create(<Button type="button" layout="primary"/>);
        expect(button.toJSON()).toMatchSnapshot();
        button = TestRenderer.create(<Button type="submit" layout="primary"/>);
        expect(button.toJSON()).toMatchSnapshot();
        button = TestRenderer.create(<Button type="button" layout="primary" className="test1"/>);
        expect(button.toJSON()).toMatchSnapshot();
        button = TestRenderer.create(<Button type="submit" layout="primary" className="test2"/>);
        expect(button.toJSON()).toMatchSnapshot();
    });

    test('renders secondary correctly', () => {
        let button = TestRenderer.create(<Button type="button" layout="secondary"/>);
        expect(button.toJSON()).toMatchSnapshot();
        button = TestRenderer.create(<Button type="submit" layout="secondary"/>);
        expect(button.toJSON()).toMatchSnapshot();
        button = TestRenderer.create(<Button type="button" layout="secondary" className="test1"/>);
        expect(button.toJSON()).toMatchSnapshot();
        button = TestRenderer.create(<Button type="submit" layout="secondary" className="test2"/>);
        expect(button.toJSON()).toMatchSnapshot();
    });

});