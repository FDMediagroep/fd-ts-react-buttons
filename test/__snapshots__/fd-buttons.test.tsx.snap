// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  className="sc-bdVaJa ZwXDY"
  type="button"
/>
`;

exports[`Button renders correctly 2`] = `
<button
  className="sc-bdVaJa ZwXDY"
  type="submit"
/>
`;

exports[`Button renders correctly 3`] = `
<button
  className="test1 sc-bdVaJa ZwXDY"
  type="button"
/>
`;

exports[`Button renders correctly 4`] = `
<button
  className="test2 sc-bdVaJa ZwXDY"
  type="submit"
/>
`;

exports[`Button renders primary correctly 1`] = `
<button
  className="primary sc-bdVaJa ZwXDY"
  type="button"
/>
`;

exports[`Button renders primary correctly 2`] = `
<button
  className="primary sc-bdVaJa ZwXDY"
  type="submit"
/>
`;

exports[`Button renders primary correctly 3`] = `
<button
  className="primary test1 sc-bdVaJa ZwXDY"
  type="button"
/>
`;

exports[`Button renders primary correctly 4`] = `
<button
  className="primary test2 sc-bdVaJa ZwXDY"
  type="submit"
/>
`;

exports[`Button renders secondary correctly 1`] = `
<button
  className="secondary sc-bdVaJa ZwXDY"
  type="button"
/>
`;

exports[`Button renders secondary correctly 2`] = `
<button
  className="secondary sc-bdVaJa ZwXDY"
  type="submit"
/>
`;

exports[`Button renders secondary correctly 3`] = `
<button
  className="secondary test1 sc-bdVaJa ZwXDY"
  type="button"
/>
`;

exports[`Button renders secondary correctly 4`] = `
<button
  className="secondary test2 sc-bdVaJa ZwXDY"
  type="submit"
/>
`;
